rclcpp/test/topic_statistics/test_subscription_topic_statistics.cpp

```diff
@@ -63,6 +77,42 @@ class TestSubscriptionTopicStatistics : public SubscriptionTopicStatistics<Callb
   }
 };
 
+/**
+ * Empty publisher node: used to publish empty messages
+ */
+class EmptyPublisher : public rclcpp::Node
+{
+public:
+  EmptyPublisher(
+    const std::string & name, const std::string & topic,
+    const std::chrono::milliseconds & publish_period = std::chrono::milliseconds{100})
+  : Node(name)
+  {
+    publisher_ = create_publisher<Empty>(topic, 10);
+    publish_timer_ = this->create_wall_timer(
+      publish_period, [this]() {
+        this->publish_message();
+      });
+  }
+
+  int get_number_published()
+  {
+    return number_published_.load();
+  }
+
+private:
+  void publish_message()
+  {
+    ++number_published_;
+    auto msg = Empty{};
+    publisher_->publish(msg);
+  }
+
+  rclcpp::Publisher<Empty>::SharedPtr publisher_;
+  std::atomic<int> number_published_{0};
```

Comment:
Does it need to be signed?

```suggestion
  std::atomic<size_t> number_published_{0};
```